test/others/criu-ns/run.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+
+import os
+import pty
+import shutil
+import subprocess
+import sys
+import time
+
+
+CRIU_BIN = "../../../criu/criu"
+CRIU_NS = "../../../scripts/criu-ns"
+
+
+def check_dumpdir():
+    if os.path.isdir("dumpdir"):
+        shutil.rmtree("dumpdir")
+    os.mkdir("dumpdir", 0o755)
+
+
+def create_pty():
+    fd_m, fd_s = pty.openpty()
+    return (os.fdopen(fd_m, "wb"), os.fdopen(fd_s, "wb"))
+
+
+def test_dump_and_restore_with_shell_job():
+    check_dumpdir()
+
+    open("running", "w").close()
+    fd_m, fd_s = create_pty()
+    pid = os.fork()
+    if pid == 0:
+        os.setsid()
+        while True:
+            if not os.access("running", os.F_OK):
+                sys.exit(0)
+            time.sleep(1)
+
+    cmd = [CRIU_NS, "dump", "-D", "dumpdir", "-v", "--shell-job",
+           "-t", str(pid), "--criu-binary", CRIU_BIN]
+    ret = subprocess.Popen(cmd).wait()
+    if ret != 0:
+        sys.exit(ret)
+
+    os.unlink("running")
+    fd_m, fd_s = create_pty()
+    pid = os.fork()
+    if pid == 0:
+        fd_m.close()
+        os.dup2(fd_s.fileno(), 0)
```

Comment:
Thanks for the review :)
`criu-ns restore` takes the control of pts `stdin`, only the `stdin`. so I left other fds as it is since they don't interfere with my test cases or the CI. Plus, if I duplicate the remaining `stdout` and `stderr` of the pts, then I'll have to alter my `criu-ns` command by adding `--log-file=` option, then I'll have print the content of the log file specified by `--log-file=` to verify the `criu-ns restore`. So, I took the simpler approach.

Is it really necessary to "use the same pts for stdout/stderr?"

Reply:
Thanks! Will do.